buildflow/io/clickhouse/strategies/clickhouse_strategies.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+import logging
+from typing import Any, Callable, Dict, Iterable, Type
+
+import clickhouse_connect
+import pandas as pd
+
+from buildflow.core.credentials import EmptyCredentials
+from buildflow.core.types.clickhouse_types import (
+    ClickhouseDatabase,
+    ClickhouseHost,
+    ClickhousePassword,
+    ClickhouseTableID,
+    ClickhouseUsername,
+)
+from buildflow.io.strategies.sink import SinkStrategy
+from buildflow.io.utils.schemas import converters
+
+_MAX_CONNECT_TRIES = 25
+
+
+class ClickhouseSink(SinkStrategy):
+    def __init__(
+        self,
+        *,
+        credentials: EmptyCredentials,
+        host: ClickhouseHost,
+        username: ClickhouseUsername,
+        password: ClickhousePassword,
+        database: ClickhouseDatabase,
+        table: ClickhouseTableID,
+    ):
+        super().__init__(credentials=credentials, strategy_id="clickhouse-sink")
+        self.client = None
+        self.database = database
+        self.table = table
+        self.host = host
+        self.username = username
+        self.password = password
+        self.connect()
+
+    async def connect(self):
```

Comment:
I missed this on the first time, but I think this method can't be async since it is called in __init__ (you can't call async function in there IIRC). The other option would be to initialize it on push if it hasn't been initialized yet. I think I would vote for that option.

Reply:
I too felt the same. By the way, what is IIRC?

Reply:
Ope sorry it's just shorthand for "if I recall correctly"

Reply:
And yes, intialized client object in the push method.